pkg/secretless/entrypoint/entrypoint.go

```diff
@@ -50,49 +53,88 @@ func StartSecretless(params *SecretlessOptions) {
 		log.Fatalln(err)
 	}
 
-	// Exit signal listener
-
-	// Coordinates processes interested in exit signals
-	exitListener := signal.NewExitListener()
-
 	// Optional Performance Profiling
-
 	handlePerformanceProfiling(params.ProfilingMode, exitListener)
 
-	// Start Services
-
-	allServices := proxyservice.NewProxyServices(cfg, availPlugins, logger, evtNotifier)
-	exitListener.AddHandler(func() {
-		fmt.Println("wait for all services signal")
-		err := allServices.Stop()
-		if err != nil {
-			// Log but but allow cleanup of other subscribers to continue.
-			log.Println(err)
-		}
-	})
+	configChangedChan, err := newConfigChangeChan(
+		params.ConfigFile,
+		params.ConfigManagerSpec,
+		params.FsWatchEnabled,
+	)
+	if err != nil {
+		log.Fatalln(err)
+	}
 
 	// Health check: Initialized
 	util.SetAppInitializedFlag()
 	util.SetAppIsLive(false)
 
-	err = allServices.Start()
-	if err != nil {
-		log.Fatalln(err)
+	var allServices internal.Service
+
+	configReload := func(cfg v2.Config) {
+		util.SetAppIsLive(false)
+		// Start Services
+		allServices = proxyservice.NewProxyServices(cfg, availPlugins, logger, evtNotifier)
+		err = allServices.Start()
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		// Health check: Live
+		util.SetAppIsLive(true)
 	}
 
-	// Health check: Live
-	util.SetAppIsLive(true)
+	exitListener.AddHandler(func() {
+		fmt.Println("Received a stop signal")
+
+		if allServices != nil {
+			err := allServices.Stop()
+			if err != nil {
+				// Log but but allow cleanup of other subscribers to continue.
+				log.Println(err)
+			}
+		}
```

Comment:
Line of sight again: I'd prefer repeating the `os.Exit`, ie
```go
if allServices == nil {
  os.Exit()
}
err := allServices.Stop()
if err != nil {
  // Log but but allow cleanup of other subscribers to continue.
  log.Println(err)
}
os.Exit()

Reply:
This seems less readable to me but i'll change it up